.github/workflows/test_and_deploy.yml

```diff
@@ -53,10 +59,17 @@ jobs:
         run: pip install tox-conda
 
       - name: Test with tox
-        uses: GabrielBB/xvfb-action@v1
-        with:
-          run: python -m tox -vv
+        run: |
+          if [ "$RUNNER_OS" == "Linux" ] ; then
+            Xvfb :99 -screen 0 1024x768x24 & sleep 3
+            herbstluftwm & sleep 1
+            python -m tox -vv
+          else
+            python -m tox -vv
+          fi
```

Comment:
why did you need to change from `xvfb-action` to this code? In napari we use this action without any problem. It will be nice to see error.

Reply:
I couldn't see a way to run the window manager on the correct display with the action, because it uses `xvfb-run` and that manages the `xvfb` process for you.

Reply:
But some test fails?

Reply:
Oh I guess it was just a syntax error with that OS detection logic I put in. I will try instead to put that logic in the action to see if it works. Something like this:

```
      - name: Test with tox (Xvfb)
        if: runner.os == "Linux"
        uses: GabrielBB/xvfb-action@v1
        with:
          run: |
            herbstluftwm & sleep 1
            python -m tox -vv

      - name: Test with tox
        if: runner.os != "Linux"
        run: python -m tox -vv
        env:
          PLATFORM: ${{ matrix.platform }}
          PYVISTA_OFF_SCREEN: True
```

Reply:
It just hangs: https://github.com/aganders3/napari-console/actions/runs/3191198869/jobs/5207212911

Maybe the wm could be run in the background by tox or even pytest (in a fixture, for example)?

Reply:
> Maybe the wm could be run in the background by tox or even pytest (in a fixture, for example)?

This seemed to work (see https://github.com/napari/napari-console/pull/23/commits/23fc6cd90ea56ce7a0b088fcfb596d2975bd2eaa) and here's the passing test run: https://github.com/aganders3/napari-console/actions/runs/3191679449/jobs/5208278101

It could use a little cleanup but let me know what you think. I appreciate your feedback on the approach.

Reply:
it looks ok